op-e2e/interop/supersystem.go

```diff
@@ -0,0 +1,622 @@
+	operatorKeys[devkeys.ProposerRole] = *proposerSecret
+	// create the batcher secret
+	batcherSecret, err := s.hdWallet.Secret(devkeys.ChainOperatorKey{
+		ChainID: l2Out.Genesis.Config.ChainID,
+		Role:    devkeys.BatcherRole,
+	})
+	require.NoError(s.t, err)
+	operatorKeys[devkeys.BatcherRole] = *batcherSecret
+	return operatorKeys
+}
+
+// newGethForL2 creates a new Geth instance for an L2 chain
+func (s *interopE2ESystem) newGethForL2(id string, l2Out *interopgen.L2Output) *geth.GethInstance {
+	jwtPath := writeDefaultJWT(s.t)
+	name := "l2-" + id
+	l2Geth, err := geth.InitL2(name, l2Out.Genesis, jwtPath)
+	require.NoError(s.t, err)
+	require.NoError(s.t, l2Geth.Node.Start())
+	s.t.Cleanup(func() {
+		_ = l2Geth.Close()
+	})
+	return l2Geth
+}
+
+// newNodeForL2 creates a new Opnode for an L2 chain
+func (s *interopE2ESystem) newNodeForL2(
+	id string,
+	l2Out *interopgen.L2Output,
+	operatorKeys map[devkeys.ChainOperatorRole]ecdsa.PrivateKey,
+	l2Geth *geth.GethInstance,
+) *opnode.Opnode {
+	logger := s.logger.New("role", "op-node-"+id)
+	p2pKey := operatorKeys[devkeys.SequencerP2PRole]
+	nodeCfg := &node.Config{
+		L1: &node.PreparedL1Endpoint{
+			Client: client.NewBaseRPCClient(endpoint.DialRPC(
+				endpoint.PreferAnyRPC,
+				s.l1.UserRPC(),
+				mustDial(s.t, logger))),
+			TrustRPC:        false,
+			RPCProviderKind: sources.RPCKindDebugGeth,
+		},
+		L2: &node.L2EndpointConfig{
+			// TODO refactoring this to a PreparedL2Endpoint,
+			//  with auth, to utilize in-process RPC would be very nice.
+			L2EngineAddr:      l2Geth.AuthRPC().RPC(),
+			L2EngineJWTSecret: testingJWTSecret,
+		},
+		Beacon: &node.L1BeaconEndpointConfig{
+			BeaconAddr: s.beacon.BeaconAddr(),
+		},
+		Driver: driver.Config{
+			SequencerEnabled: true,
+		},
+		Rollup: *l2Out.RollupCfg,
+		P2PSigner: &p2p.PreparedSigner{
+			Signer: p2p.NewLocalSigner(&p2pKey)},
+		RPC: node.RPCConfig{
+			ListenAddr:  "127.0.0.1",
+			ListenPort:  0,
+			EnableAdmin: true,
+		},
+		Supervisor: &node.SupervisorEndpointConfig{
+			SupervisorAddr: s.supervisor.RPC(),
+		},
+		P2P:                         nil, // disabled P2P setup for now
+		L1EpochPollInterval:         time.Second * 2,
+		RuntimeConfigReloadInterval: 0,
+		Tracer:                      nil,
+		Sync: sync.Config{
+			SyncMode:                       sync.CLSync,
+			SkipSyncStartCheck:             false,
+			SupportsPostFinalizationELSync: false,
+		},
+		ConfigPersistence: node.DisabledConfigPersistence{},
+	}
+	opNode, err := opnode.NewOpnode(logger.New("service", "op-node"),
+		nodeCfg, func(err error) {
+			s.t.Error(err)
+		})
+	require.NoError(s.t, err)
+	s.t.Cleanup(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel() // force-quit
+		_ = opNode.Stop(ctx)
+	})
+	return opNode
+}
+
+// newProposerForL2 creates a new Proposer for an L2 chain
+// it is currently unused, as the generated world does not have a DisputeGameFactoryProxy
+func (s *interopE2ESystem) _(
```

Comment:
no method name?

Reply:
Yeah the linter didn't like that it was unused so I just removed them for now. The comment indicates what this function name would be. Felt silly to delete or comment the whole function, but open to suggestion.